sims/verilator/Makefile

```diff
@@ -47,67 +47,133 @@ debug: $(sim_debug)
+VMODE := \
 	-O3 \
-	-CFLAGS "$(CXXFLAGS) -DTEST_HARNESS=V$(VLOG_MODEL) -DVERILATOR" \
-	-CFLAGS "-I$(build_dir) -include $(build_dir)/$(long_name).plusArgs -include $(build_dir)/verilator.h" \
-	-LDFLAGS "$(LDFLAGS)" \
-	$(RISCV)/lib/libfesvr.a \
-	$(dramsim_lib)
-
-# default flags added for ariane
-ARIANE_VERILATOR_FLAGS = \
+	--x-assign fast \
+	--x-initial fast \
+	--assert \
+	--output-split 10000 \
+	--output-split-cfuncs 10000
+
+# default flags added for ariane (-Wno-fatal needed for -Wall to not cause
+# a crash, since 1000s of warnings are generated)
+VERILOG_IP_VERILATOR_FLAGS := \
 	--unroll-count 256 \
-	-Werror-PINMISSING \
-	-Werror-IMPLICIT \
 	-Wno-PINCONNECTEMPTY \
 	-Wno-ASSIGNDLY \
 	-Wno-DECLFILENAME \
 	-Wno-UNUSED \
 	-Wno-UNOPTFLAT \
 	-Wno-BLKANDNBLK \
-	-Wno-style \
-	-Wall
+	-Wno-fatal
+
+# normal flags used for chipyard builds (that are incompatible with vlog ip aka ariane)
+CHIPYARD_VERILATOR_FLAGS :=
 
-# normal flags used for chipyard builds (that are incompatible with ariane)
-CHIPYARD_VERILATOR_FLAGS = \
-	--assert
+# options dependent on whether ariane/NVDLA or chipyard is used
+# NOTE: defer the evaluation of this until it is used!
+PLATFORM_OPTS = $(shell \
+	if grep -qiP "module\s+(Ariane|NVDLA)" $(build_dir)/*.*v; \
+	then echo "$(VERILOG_IP_VERILATOR_FLAGS)"; \
+	else echo "$(CHIPYARD_VERILATOR_FLAGS)"; fi)
 
 # Use --timescale to approximate timescale behavior of pre-4.034
 TIMESCALE_OPTS := $(shell verilator --version | perl -lne 'if (/(\d.\d+)/ && $$1 >= 4.034) { print "--timescale 1ns/1ps"; }')
+
+# see: https://github.com/ucb-bar/riscv-mini/issues/31
+MAX_WIDTH_OPTS = $(shell verilator --version | perl -lne 'if (/(\d.\d+)/ && $$1 > 4.016) { print "--max-num-width 1048576"; }')
+
+PREPROC_DEFINES := \
+	+define+PRINTF_COND=\$$c\(\"verbose\",\"\&\&\"\,\"done_reset\"\) \
+	+define+STOP_COND=\$$c\(\"done_reset\"\)
+
 VERILATOR_NONCC_OPTS = \
+	$(RUNTIME_PROFILING_VFLAGS) \
+	$(VMODE) \
+	$(PLATFORM_OPTS) \
 	$(TIMESCALE_OPTS) \
+	$(MAX_WIDTH_OPTS) \
+	$(PREPROC_DEFINES) \
 	--top-module $(VLOG_MODEL) \
 	--vpi \
-	-Wno-fatal \
-	$(shell if ! grep -iq "module.*ariane" $(build_dir)/*.*v; then echo "$(CHIPYARD_VERILATOR_FLAGS)"; else echo "$(ARIANE_VERILATOR_FLAGS)"; fi) \
-	--output-split 10000 \
-	--output-split-cfuncs 100 \
-	--max-num-width 1048576 \
 	-f $(sim_common_files) \
 	$(sim_vsrcs)
 
-VERILATOR_DEFINES = \
-	+define+PRINTF_COND=\$$c\(\"verbose\",\"\&\&\"\,\"done_reset\"\) \
-	+define+STOP_COND=\$$c\(\"done_reset\"\)
+#----------------------------------------------------------------------------------------
+# gcc configuration/optimization
+#----------------------------------------------------------------------------------------
+# -flto slows down compilation on small-memory and breaks on firesim-manager
+CMODE := -O3 -fbranch-probabilities -march=native
+
+VERILATOR_CXXFLAGS = \
+	$(CXXFLAGS) \
+	$(RUNTIME_PROFILING_CFLAGS) \
+	$(TRACING_CFLAGS) \
+	$(CMODE) \
+	-std=c++11 \
```

Comment:
Why do all of these not get pulled into any of the above variables, or another variable?

Reply:
I think these are just general CFLAGS and thus they aren't profiling, tracing, or optimization flags...